Firmware/first_lay_cal.cpp

```diff
@@ -210,41 +224,37 @@ void lay1cal_meander(float layer_height, float extrusion_width)
 //! @param i iteration
 void lay1cal_square(uint8_t step, float layer_height, float extrusion_width)
 {
-    const float long_length = 20;
-    const float short_length = spacing(layer_height, extrusion_width);
-    const float long_extrusion = count_e(layer_height, extrusion_width, long_length);
-    const float short_extrusion = count_e(layer_height, extrusion_width, short_length);
-    static const char fmt1[] PROGMEM = "G1 X%d Y%-.2f E%-.3f";
+    const float Y_spacing = spacing(layer_height, extrusion_width);
+    const float long_extrusion = count_e(layer_height, extrusion_width, square_width);
+    const float short_extrusion = count_e(layer_height, extrusion_width, Y_spacing);
 
     for (uint8_t i = step; i < step+4; ++i)
     {
-        enquecommandf_P(fmt1, 70, (35 - i*short_length * 2), long_extrusion);
-        enquecommandf_P(fmt1, 70, (35 - (2 * i + 1)*short_length), short_extrusion);
-        enquecommandf_P(fmt1, 50, (35 - (2 * i + 1)*short_length), long_extrusion);
-        enquecommandf_P(fmt1, 50, (35 - (i + 1)*short_length * 2), short_extrusion);
+        enquecommandf_P(extrude_fmt_X, square_width*invert, long_extrusion);
+        enquecommandf_P(extrude_fmt_Y, -Y_spacing*invert, short_extrusion);
+        enquecommandf_P(extrude_fmt_X, -square_width*invert, long_extrusion);
+        enquecommandf_P(extrude_fmt_Y, -Y_spacing*invert, short_extrusion);
     }
 }
 
 void lay1cal_finish(bool mmu_enabled)
 {
-    static const char cmd_cal_finish_1[] PROGMEM = "G1 E-0.075 F2100"; //retract
-    static const char cmd_cal_finish_2[] PROGMEM = "M104 S0"; // turn off temperature
-    static const char cmd_cal_finish_3[] PROGMEM = "M140 S0"; // turn off heatbed
-    static const char cmd_cal_finish_4[] PROGMEM = "G1 Z10 F1300"; //lift Z
-    static const char cmd_cal_finish_5[] PROGMEM = "G1 X10 Y180 F4000"; //Go to parking position
+    static const char cmd_cal_finish_2[] PROGMEM = "G1E-0.075F2100"; //retract
+    static const char cmd_cal_finish_3[] PROGMEM = "G1Z10F1300"; //lift Z
+    static const char cmd_cal_finish_4[] PROGMEM = "G1X10Y180F4000"; //Go to parking position
 
     static const char * const cmd_cal_finish[] PROGMEM =
     {
             MSG_M107, // turn off printer fan
-            cmd_cal_finish_1,
             cmd_cal_finish_2,
             cmd_cal_finish_3,
-            cmd_cal_finish_4,
-            cmd_cal_finish_5
+            cmd_cal_finish_4
     };
 
+    enquecommand_P(MSG_G90);
     lay1cal_common_enqueue_loop(cmd_cal_finish, (sizeof(cmd_cal_finish)/sizeof(cmd_cal_finish[0])));
 
     if (mmu_enabled) enquecommand_P(MSG_M702); //unload from nozzle
+    disable_heater();
```

Comment:
Note that MMU M702 needs the heater to be on to complete the unload. The heaters are already turned off above with  `M140 S0` (See `cmd_cal_finish_2`) and `M104 S0` (See `cmd_cal_finish_3`)

We can then also avoid including `#include "temperature.h"`

Now that I look at the code, the `M702` g-code should actually come before `M104 S0`

Reply:
Changing to `disable_heater();` saves few additional bytes.

Reply:
I agree with @gudnimg. We should not disable the heaters before the M702. Moreover, directly calling disable_heater bypasses the cmdqueue, so stuff ends up executing out of order. Please switch back to M104 and M140, but after the M702 stuff. I know it will add back some bytes, but I would rather have that than mix and match cmdqueue and non-cmdqueue actions in the automation and cause issues with the M702 unload.

Also grrrrrr the MK3SMMU3 prusaslicer profiles also do this out of order :/
![image](https://github.com/user-attachments/assets/fb6f5f44-f1c2-4505-8b46-c6df330ed7b2)

Reply:
I'd like to see the changes from #4813 in this PR to address this issue.

Reply:
I’ll prepare a PR on @3d-gussner ‘s fork soon

Reply:
Merged. Should be resolved now. @3d-gussner please check this in the sim.